src/pylexibank/cldf.py

```diff
@@ -102,6 +102,57 @@ def cognate_id(self, kw):
     def tokenize(self, item, string):
         if self.dataset.tokenizer:
             return self.dataset.tokenizer(item, string)
+    
+    def add_form(self, **kw):
+        """
+        :return: dict with the newly created form
+        """
+        kw_ = kw.copy()
+
+        # Do we have morpheme segmentation on top of phonemes?
+        with_morphemes = '+' in self['FormTable', 'Segments'].separator
+
+        language, concept, value, form = [kw_.get(v) for v in [
+            'Language_ID', 'Parameter_ID', 'Value', 'Form']]
+        # check for required kws
+        if language is None or concept is None or value is None \
```

Comment:
Maybe `if not all([language, concept, value, form]):` ?

Reply:
Ah, doesn't work: if the form is empty, it should not raise an error, just not add it, but if it is not passed as an argument, it'll be none, and then it should raise an error. So the current is capturing this, all cannot handle it in the variants I tested.

Reply:
Hm, not sure if I am following, your code is raising the error if only the `form` is empty:

```
>>> l, c, v, f = 1, 2, 3, None
>>> if l is None or c is None or v is None or f is None:
...   print("err")
err
```

Could you give the tests you are using?

Reply:
My answer did not come through... So here again quickly: If either of the major keywords is not passed, it should raise an error. We could also just make this a function with ordered arguments. While something like "form" can in fact be False by truth value, it should never be None, as None means it was not passed, so it should through the error in this case only. all() or any() check for truth values, not for None.

Reply:
Yes, you could technically solve by checking for `is None` or similar, but it would go against my original idea of having a shorter and faster to understand logic. Better keep it this way.